test/unit/io/sequence_file/sequence_file_integration_test.cpp

```diff
@@ -135,3 +135,30 @@ TEST(integration, convert_fastq_to_fasta)
     fout.get_stream().flush();
     EXPECT_EQ(reinterpret_cast<std::ostringstream&>(fout.get_stream()).str(), fasta_out);
 }
+
+TEST(integration, tricky_id)
+{
+    // The read/written part of the id should start with '> ' for indicating a sequence.
+    // The sequence id is '>TEST 1'.
+    std::string const input
+    {
+        "> >TEST 1\n"
+        "ACGT\n"
+    };
```

Comment:
```suggestion
    // The sequence id is '>TEST 1'.
    std::string const input
    {
        "> >TEST 1\n"
        "ACGT\n"
    };
```

Could be removed if test name is changed, otherwise:
```
The sequence name should start with `>`, which usually marks a line as containing the ID.
```